influxdata/tokenizer.go

```diff
@@ -111,24 +207,28 @@ func (t *Tokenizer) readMore() {
 	n := cap(t.buf) - len(t.buf)
 	if n < minRead {
 		// There's not enough available space at the end of the buffer to read into.
-		if n+t.r >= minRead {
+		if t.r0+n >= minRead {
 			// There's enough space when we take into account already-used
 			// part of buf, so slide the data to the front.
-			copy(t.buf, t.buf[t.r:])
-			t.buf = t.buf[:len(t.buf)-t.r]
+			copy(t.buf, t.buf[t.r0:])
+			t.buf = t.buf[:len(t.buf)-t.r1]
+			t.r1 -= t.r0
+			t.r0 = 0
 		} else {
 			// We need to grow the buffer. Note that we don't have to copy
-			// the unused part of the buffer (t.buf[:t.r]).
+			// the unused part of the buffer (t.buf[:t.r0]).
 			// TODO provide a way to limit the maximum size that
 			// the buffer can grow to.
-			used := len(t.buf) - t.r
+			used := len(t.buf) - t.r0
 			n1 := cap(t.buf) * 2
 			if n1-used < minRead {
 				n1 = used + minRead
 			}
 			buf1 := make([]byte, used, n1)
-			copy(buf1, t.buf[t.r:])
+			copy(buf1, t.buf[t.r0:])
 			t.buf = buf1
+			t.r0 = 0
+			t.r1 = used
```

Comment:
I believe this is incorrect. It is advancing the read (past tense) pointer too far.

```
t.r1 -= t.r0
t.r0 = 0
```

Reply:
yes! excellent catch, thanks! i'll add a test.